src/utilities.js

```diff
@@ -250,3 +251,92 @@ export function getResults(request) {
     });
   };
 }
+
+export function getTrendingResults(request) {
+  return (searchType, obj) => {
+    const map = {
+      'Daily trends': {
+        path: '/trends/api/dailytrends',
+        extraParams: {
+          ed: convertDateToString(obj.trendDate, false, true),
+          ns: obj.ns,
+        },
+      },
+      'Real time trends': {
+        path: '/trends/api/realtimetrends',
+        extraParams: {
+          fi: 0,
+          fs: 0,
+          ri: 300, // # of trending stories IDs returned
+          rs: 20,
+          sort: 0,
+        },
+      },
+    };
+
+    let options = {
+      method: 'GET',
+      host: 'trends.google.com',
+      path: map[searchType].path,
+      qs: {
+        hl: obj.hl,
+        tz: obj.timezone,
+        geo: obj.geo,
+        cat: obj.category,
+      },
+    };
+
+    if (obj.agent) options.agent = obj.agent;
+
+    Object.assign(options.qs, options.qs, map[searchType].extraParams);
```

Comment:
I'm a little confused about this line, does this mutate the original `options.qs` and adds `extraParams`?

I prefer to use the [spread operator](https://github.com/airbnb/javascript#objects--rest-spread). Can you do something like:
```
options.qs = {...options.qs, ...map[searchType].extraParams};
```

Reply:
Or maybe make `qs` it's own object above:
```
const qs = {
        hl: obj.hl,
        tz: obj.timezone,
        geo: obj.geo,
        cat: obj.category,
};
```
and then just spread `qs` into each `extraParams`, then you don't even need this line

Reply:
So above would be like

```
     'Daily trends': {
        path: '/trends/api/dailytrends',
        extraParams: {
          ed: convertDateToString(obj.trendDate, false, true),
          ns: obj.ns,
          ...qs
        },
```

Reply:
Hi. Yes for mutation purposes. Ill use the one line change that you suggested just because it is the clearest in terms of behavior changes. Done.